Bottom: 0c7d9ceab0d5df15acf756ee788c788d99747f7d
Top:    2158c8de70f0c5c65ab2c04bb48616abcce5846f
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-07-21 20:23:12 +0800

vhost: prevent being kicked after vhost_disable_notify() for event idx

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index e4613a3..e9eea3b 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -2386,7 +2386,16 @@ void vhost_disable_notify(struct vhost_dev *dev, struct vhost_virtqueue *vq)
 		if (r)
 			vq_err(vq, "Failed to enable notification at %p: %d\n",
 			       &vq->used->flags, r);
+	} else {
+		r = vhost_update_avail_event(vq, vq->last_avail_idx);
+		if (r) {
+			vq_err(vq, "Failed to update avail event index at %p: %d\n",
+			       vhost_avail_event(vq), r);
+			return;
+		}
 	}
+
+	return;
 }
 EXPORT_SYMBOL_GPL(vhost_disable_notify);
